src/wp-content/themes/twentytwentyone/assets/js/primary-navigation.js

```diff
@@ -173,17 +173,19 @@ function twentytwentyoneExpandSubMenu( el ) { // jshint ignore:line
 		 * @since Twenty Twenty-One 1.1
 		 */
 		document.addEventListener( 'click', function( event ) {
-			// If target onclick is <a> with # within the href attribute
-			if ( event.target.hash && event.target.hash.includes( '#' ) ) {
-				wrapper.classList.remove( id + '-navigation-open', 'lock-scrolling' );
-				twentytwentyoneToggleAriaExpanded( mobileButton );
-				// Wait 550 and scroll to the anchor.
-				setTimeout(function () {
-					var anchor = document.getElementById(event.target.hash.slice(1));
-					if ( anchor ) {
-						anchor.scrollIntoView();
-					}
-				}, 550);
+			if ( document.getElementById( 'site-navigation' ).contains( event.target ) ) {
```

Comment:
I'm not sure attaching a click event to the document (line 175) and then checking if the target is a navigation link is ideal. Events on the document are a bit too 'global' and should be avoided when possible. A better option would be selecting all the navigation links and loop through them to attach a click event only to the links with a page fragment identifier (a 'hash').